ext/liquid_c/tokenizer.c

```diff
@@ -127,6 +128,10 @@ void tokenizer_next(tokenizer_t *tokenizer, token_t *token)
             cursor++;
         }
     }
+    if(token->type == TOKEN_RAW && tokenizer->trim_whitespace) {
+        token->trim_whitespace = 1;
+    }
+    tokenizer->trim_whitespace = ((token->type == TOKEN_VARIABLE || token->type == TOKEN_TAG) && token->str[token->length - 3] == '-');
```

Comment:
I don't see why you are doing these checks down here when you could do them in context where token->type is set.

That would also allow you to easily detect if the next token should be trimmed, since the function finds the end of a raw string by finding the start of the next tag or variable.  That means you could avoid having to construct a ruby string that needs to be right stripped.  You could store that as two flags on the token for whether the raw tag should be left and/or right trimmed.